src/signing/transaction-manager.js

```diff
@@ -0,0 +1,192 @@
+import config from "src/../config"
+import { getSigner, cancelSign, signQueue } from "./signer"
+import { getGraphqlHost } from "scripts/url"
+import { getFingerprint } from "scripts/fingerprint"
+import { getMessage } from "./message-creator.js"
+import gql from "graphql-tag"
+
+const txFetchOptions = fingerprint => ({
+  method: "POST",
+  headers: {
+    "Content-Type": "application/json",
+    fingerprint
+  }
+})
+
+export default class TransactionManager {
+  constructor(apolloClient) {
+    this.apollo = apolloClient
+  }
+
+  async broadcastAPIRequest(payload) {
+    const fingerprint = await getFingerprint()
+    const options = {
+      ...txFetchOptions(fingerprint),
+      body: JSON.stringify({ payload })
+    }
+    const graphqlHost = getGraphqlHost()
+
+    return fetch(`${graphqlHost}/transaction/broadcast`, options).then(result =>
+      result.json()
+    )
+  }
+
+  async getSignQueue(submitType) {
+    return signQueue(submitType)
+  }
+
+  async cancel({ userAddress, networkId }, submitType) {
+    return await cancelSign(submitType, {
+      address: userAddress,
+      network: networkId
+    })
+  }
+
+  async getCosmosTransactionData({
+    memo,
+    gasEstimate,
+    gasPrice,
+    senderAddress,
+    network
+  }) {
+    const {
+      data: {
+        signingInfo: { accountNumber, sequence }
+      }
+    } = await this.apollo.query({
+      query: gql`
+        query signingInfo($networkId: String!, $senderAddress: String!) {
+          overview(networkId: $networkId, address: $senderAddress) {
+            accountInformation {
+              accountNumber
+              sequence
+            }
+          }
+        }
+      `,
+      variables: { networkId: network.id, senderAddress },
+      fetchPolicy: "network-only"
+    })
+    const { Coin } = await import("./networkMessages/cosmos-hub-mainnet")
+
+    return {
+      accountNumber,
+      sequence,
+      chainId: network.chain_id,
+      gasEstimate: String(gasEstimate),
+      gasPrices: [Coin(gasPrice, network.coinLookup)],
+      memo
+    }
+  }
+
+  async createSignBroadcast({
+    messageType,
+    message,
+    transactionData,
+    senderAddress,
+    network,
+    signingType,
+    password
+  }) {
+    const signer = await getSigner(
+      signingType,
+      {
+        address: senderAddress,
+        password,
+        network: network.id,
+        networkType: network.network_type
+      },
+      config // only needed for Ledger
+    )
+
+    const broadcastableObject = await createAndSign(
```

Comment:
`createAndSign´ together with `getSigner` can be used in the extension when we send to the extension

```
    messageType,
    message,
    transactionData,
    senderAddress,
    network,
```